lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -377,8 +373,8 @@ def test_http_exception(
         if not self.elastic_endpoint:
             pytest.skip("skipping " + self.test_http_exception.__name__)
 
-        if self.use_index_from_metadata:
-            index = self.build_index_from_metadata(self.RESULT_DATA_SAMPLE)
+        if self.index_from_metadata:
+            index = self.build_index_from_metadata()
         else:
             index = self.build_index(
                 server_config,
```

Comment:
This `if`/`else` is repeated several times, and that stands out in the pattern of your changes. It could really be refactored into a new superclass method, which would streamline the test cases a bit.

Reply:
Okay created an issue to track https://github.com/distributed-system-analysis/pbench/issues/2582